src/types/output.rs

```diff
@@ -21,6 +24,33 @@ pub struct OutputLayout {
 }
 
 impl Output {
+    pub unsafe fn set_user_data(&mut self, layout: Rc<RefCell<OutputLayout>>) {
```

Comment:
We should change this it an internal struct, e.g called `OutputDate` or `OutputState` so that it's easier to extend